src/main/java/pl/connectis/Spotify_API_CLI/SpotifyAPICLR.java

```diff
@@ -0,0 +1,74 @@
+package pl.connectis.Spotify_API_CLI;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.cli.*;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import pl.connectis.Spotify_API_CLI.APICalls.APICallerFactory;
+import pl.connectis.Spotify_API_CLI.authorization.AuthorizationStrategy;
+import pl.connectis.Spotify_API_CLI.authorization.Token;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+@Component
+@Slf4j
+@Profile({"!test"})
+public class SpotifyAPICLR implements CommandLineRunner {
+
+    //    @Value("${authorizationMethod}")
+//    static private String authorizationMethod;
+    @Qualifier("${authorizationMethod}")
+    private final AuthorizationStrategy authorization;
+    private final APICallerFactory apiCallerFactory;
+    private Token token;
+
+
+    public SpotifyAPICLR(AuthorizationStrategy authorization, APICallerFactory apiCallerFactory, Token token) {
+        this.authorization = authorization;
+        this.apiCallerFactory = apiCallerFactory;
+        this.token = token;
+    }
+
+    public void run(String... args) {
+
+        log.info("args: {}", Arrays.toString(args));
+
+        Options options = new CLIOptionsBuilder().build();
+        CommandLineParser CLIParser = new DefaultParser();
+        CommandLine cmd;
+        try {
+            cmd = CLIParser.parse(options, args);
+        } catch (ParseException ex) {
+            log.error("Parse error: " + ex.getMessage());
+            ex.printStackTrace();
+            return;
+        }
+
+
+        if (!cmd.hasOption("a") && (!token.hasToken() || !token.isTokenValid())) {
+            authorize(authorization);
+        }
+
+        if (cmd.hasOption("a")) {
+            authorize(authorization);
+
+        } else {
+            String ids = cmd.getOptionValue(cmd.getOptions()[0].getOpt());
+            log.info("Parsed ids: {}", ids);
+            apiCallerFactory.getCaller(cmd.getOptions()[0].getLongOpt()).call(ids);
```

Comment:
Już tutaj bym sparsował input użytkownika żeby dalej było wiadomo czy mamy 1 id czy kilka idków

Reply:
Ja to rozpoznaję w metodzie `call(ids)` odpowiedniego `ApiCaller` patrzac czy jest w argumencie `,` czy go nie ma.

Reply:
jedyne co sugeruje to przenieść ten kod tutaj. Im szybciej będziesz operował na konkretnych typach (String lub kolekcja w zaleznosci czy przecinek jest) tym trudniej o późniejszą pomyłkę